util/k8s/k8sApplicationService.go

```diff
@@ -1110,3 +1141,216 @@ func (impl *K8sApplicationServiceImpl) FetchConnectionStatusForCluster(k8sClient
+	return err
+}
+
+func (impl *K8sApplicationServiceImpl) generateDebugContainer(pod *corev1.Pod, req cluster.EphemeralContainerRequest) (*corev1.Pod, *corev1.EphemeralContainer, error) {
+	copied := pod.DeepCopy()
+	ephemeralContainer := &corev1.EphemeralContainer{}
+	if req.AdvancedData != nil {
+		err := json.Unmarshal([]byte(req.AdvancedData.Manifest), ephemeralContainer)
+		if err != nil {
+			impl.logger.Errorw("error occurred in unMarshaling advanced ephemeral data", "err", err, "advancedData", req.AdvancedData.Manifest)
+			return copied, ephemeralContainer, err
+		}
+		if ephemeralContainer.TargetContainerName == "" || ephemeralContainer.Name == "" || ephemeralContainer.Image == "" {
+			return copied, ephemeralContainer, errors.New("containerName,targetContainerName and image cannot be empty")
+		}
+	} else {
+		ephemeralContainer = &corev1.EphemeralContainer{
+			EphemeralContainerCommon: corev1.EphemeralContainerCommon{
+				Name:                     req.BasicData.ContainerName,
+				Env:                      nil,
+				Image:                    req.BasicData.Image,
+				ImagePullPolicy:          corev1.PullIfNotPresent,
+				Stdin:                    true,
+				TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+				TTY:                      true,
+			},
+			TargetContainerName: req.BasicData.TargetContainerName,
+		}
+	}
+	ephemeralContainer.Name = ephemeralContainer.Name + "-" + util2.Generate(5)
+	scriptCreateCommand := fmt.Sprintf("echo 'while true; do sleep 600; done;' > %s-devtron.sh", ephemeralContainer.Name)
+	scriptRunCommand := fmt.Sprintf("sh %s-devtron.sh", ephemeralContainer.Name)
+	ephemeralContainer.Command = []string{"sh", "-c", scriptCreateCommand + " && " + scriptRunCommand}
+	copied.Spec.EphemeralContainers = append(copied.Spec.EphemeralContainers, *ephemeralContainer)
+	ephemeralContainer = &copied.Spec.EphemeralContainers[len(copied.Spec.EphemeralContainers)-1]
+	return copied, ephemeralContainer, nil
+
+}
+
+func (impl *K8sApplicationServiceImpl) TerminatePodEphemeralContainer(req cluster.EphemeralContainerRequest) (bool, error) {
+	terminalReq := &terminal.TerminalSessionRequest{
+		PodName:       req.PodName,
+		ClusterId:     req.ClusterId,
+		Namespace:     req.Namespace,
+		ContainerName: req.BasicData.ContainerName,
+	}
+	container, err := impl.ephemeralContainerRepository.FindContainerByName(terminalReq.ClusterId, terminalReq.Namespace, terminalReq.PodName, terminalReq.ContainerName)
+	if err != nil {
+		impl.logger.Errorw("error in finding ephemeral container in the database", "err", err, "ClusterId", terminalReq.ClusterId, "Namespace", terminalReq.Namespace, "PodName", terminalReq.PodName, "ContainerName", terminalReq.ContainerName)
+		return false, err
+	}
+	if container == nil {
+		return false, errors.New("externally created ephemeral containers cannot be removed")
+	}
+	containerKillCommand := fmt.Sprintf("kill -16 $(pgrep -f '%s-devtron' -o)", terminalReq.ContainerName)
```

Comment:
what about `pgrep` support. Have we checked ??

Reply:
i have checked with given image options and also with alpine image, it's working